client/src/components/PostList.jsx

```diff
@@ -70,6 +72,15 @@ const PostList = () => {
     // }
   }, [filterSettings, posts]);
 
+  const handleUnapproveClick = (event) => {
+    unapprovePost(event.target.value)
+    getPublicPosts().then((posts) => {
```

Comment:
add the `getPublicPosts` call in `.then` of the `unapprovePost` call, otherwise they may not be executed in order since `unapprovePost` is asynchronous.